pkg/agent/utils_test.go

```diff
@@ -743,6 +746,39 @@ var _ = Describe("Test GetOrderedKubeletConfigFlagString", func() {
 		actualStr := GetOrderedKubeletConfigFlagString(config)
 		Expect(expectedStr).To(Equal(actualStr))
 	})
+
+	//https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/#kubelet-configuration-merging-order
+	// [we are producing this -> command line flags] > drop in config file > kubelet config
+	It("should return expected kubelet command line flags when a config file is being used, following overriding rules", func() {
+		cs := &datamodel.ContainerService{
+			Location: "southcentralus",
+			Type:     "Microsoft.ContainerService/ManagedClusters",
+			Properties: &datamodel.Properties{
+				CustomConfiguration: &datamodel.CustomConfiguration{
+					KubernetesConfigurations: map[string]*datamodel.ComponentConfiguration{
+						"kubelet": {
+							Config: map[string]string{
+								"--seccomp-default": "true",
+							},
+						},
+					},
+				},
+			},
+		}
+		k := map[string]string{
+			"--node-labels": "topology.kubernetes.io/region=southcentralus",
+		}
+
+		ap := &datamodel.AgentPoolProfile{
+			CustomKubeletConfig: &datamodel.CustomKubeletConfig{
+				SeccompDefault: to.BoolPtr(false),
+			},
+		}
+
+		expectedStr := "--node-labels=topology.kubernetes.io/region=southcentralus --seccomp-default=true "
+		actualStr := GetOrderedKubeletConfigFlagString(k, cs, ap, true)
```

Comment:
Seems this function got updated to take a datamodel, I'd just do something like 
```
dm := datamodel.NodeBootstrappingConfiguration{
			KubeletConfig:    k,
			ContainerService: cs,
			AgentPoolProfile: ap,
		}
actualStr := GetOrderedKubeletConfigFlagString(&dm)
```

Reply:
done.. rebased and resolved merge conflicts